src/server/game/AI/CoreAI/PetAI.cpp

```diff
@@ -462,7 +461,7 @@ void PetAI::DoAttack(Unit* target, bool chase)
             ClearCharmInfoFlags();
             me->GetCharmInfo()->SetIsCommandAttack(oldCmdAttack); // For passive pets commanded to attack so they will use spells
             me->GetMotionMaster()->Clear();
-            me->GetMotionMaster()->MoveChase(target);
+            me->GetMotionMaster()->MoveChase(target, me->GetPetChaseDistance());
```

Comment:
if the me->GetPetChaseDistance() > 5  and the target is not facing pet.  use MoveChase the pet will move long way to facing target  

example:  you and your pet is behind of the target " Training Dummy" it can not turn to facing you 
if GetPetChaseDistance() >= 40  
the pet will move long way from you to facing target..you and your pet's distance will  > 80